process/block/preprocess/transactions.go

```diff
@@ -731,11 +740,24 @@ func (txs *transactions) getAllTxsFromMiniBlock(
 	return txsSlice, txHashes, nil
 }
 
+func (txs *transactions) getRemainingGasPerBlock() uint64 {
+	gasConsumed := txs.gasHandler.TotalGasConsumed()
+	maxGasPerBlock := txs.economicsFee.MaxGasLimitPerBlock(txs.shardCoordinator.SelfId())
+	gasBandwidth := uint64(0)
+	if gasConsumed < maxGasPerBlock {
+		gasBandwidth = maxGasPerBlock - gasConsumed
+	}
+	return gasBandwidth
+}
+
 // CreateAndProcessMiniBlocks creates miniblocks from storage and processes the transactions added into the miniblocks
 // as long as it has time
-func (txs *transactions) CreateAndProcessMiniBlocks(haveTime func() bool) (block.MiniBlockSlice, error) {
+func (txs *transactions) CreateAndProcessMiniBlocks(haveTime func() bool, randomness []byte) (block.MiniBlockSlice, error) {
```

Comment:
On the scheduled SC calls feature, we had to push the header being processed to the transaction preprocessor, and the randomness will be available from the header as well, this is why I chose to temporarily push the randomness.
When merging the feature to dev, this will be replaced with header, and the randomness will be taken from it.

Reply:
even on scheduled. this txsPreprocessor could have simply access to chainHandler - and take everything from these.

Reply:
Chain handler does not help with the block currently being processed.

Reply:
blockchainHook has access to current Header.

Reply:
will be refactored after merging scheduled sc calls feature.